lib/puppet/indirector/request.rb

```diff
@@ -185,7 +185,11 @@ def do_request(srv_service=:puppet, default_server=nil, default_port=nil, &block
     return yield(self) if !self.server.nil?
 
     if Puppet.settings[:use_srv_records]
-      Puppet::Network::Resolver.each_srv_record(Puppet.settings[:srv_domain], srv_service) do |srv_server, srv_port|
+      # We may want to consider not creating a new resolver here
+      # every request eventually, to take advantage of the resolver's
+      # caching behavior.
+      resolver = Puppet::Network::Resolver.new
```

Comment:
Should this be in the context?

Reply:
If you check on the outdated comment from Josh on this PR, he decided we should not use the caching for the legacy code at this point, and asked me to move it out of the context to be created directly in the client and on-request here.